### PR TITLE
Prototype GMD configuration with plugin

### PR DESCRIPTION
**Pull request by utzcoz**
Fix https://github.com/robolectric/robolectric/issues/7325.

### CODE REVIEW

**Comment by utzcoz** [comment]
Special cc @yuuki3655, I have prototype defining GMD configuration to a special plugin in project's `buildSrc`, maybe you have interested into it. Also cc @hellosagar @ZSmallX  for extra reviewing.

**Review by hoisie — APPROVED** [review]

**Comment by utzcoz** [comment]
Rebased and merged. cc @hoisie .

**Comment by hellosagar** [comment]
Code looks clean. Saw this plugin code for the first time, looks interesting

File: buildSrc/src/main/groovy/org/robolectric/gradle/GradleManagedDevicePlugin.groovy [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import com.android.build.api.dsl.ManagedVirtualDevice
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+
+class GradleManagedDevicePlugin implements Plugin<Project> {
+    @Override
+    void apply(Project project) {
+        project.android.testOptions {
+            devices {
```
Comment (yuuki3655) [inline buildSrc/src/main/groovy/org/robolectric/gradle/GradleManagedDevicePlugin.groovy:11]: FYI: This has been moved to `project.android.testOptions.managedDevices.devices`.

File: buildSrc/src/main/groovy/org/robolectric/gradle/GradleManagedDevicePlugin.groovy [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+                    apiLevel = 29
+                    systemImageSource = "aosp"
+                    abi = "x86"
+                }
+                // ./gradlew -Pandroid.sdk.channel=3 nexusOneApi32DebugAndroidTest
+                nexusOneApi32(ManagedVirtualDevice) {
+                    device = "Nexus One"
+                    apiLevel = 32
+                    systemImageSource = "google"
+                    abi = "x86_64"
```
Comment (yuuki3655) [inline buildSrc/src/main/groovy/org/robolectric/gradle/GradleManagedDevicePlugin.groovy:24]: This property is replaced with `require64Bit = true`

File: buildSrc/src/main/groovy/org/robolectric/gradle/GradleManagedDevicePlugin.groovy (hunk 1 above)
Comment (utzcoz) [reply buildSrc/src/main/groovy/org/robolectric/gradle/GradleManagedDevicePlugin.groovy:11]: Hi @yuuki3655 , thanks for reviewing and suggestions. Are those changes are included from AGP 7.3? I have sent a draft PR to bump AGP to 7.3, and migrate some GMD properties to new spec: https://github.com/robolectric/robolectric/pull/7384. Could you help to review that PR? Thanks.

**Pull request merged**